test/e2e/storage/vsphere/vsphere_utils.go

```diff
@@ -467,3 +474,97 @@ func getVSphere(c clientset.Interface) (*vsphere.VSphere, error) {
 func GetVSphere(c clientset.Interface) (*vsphere.VSphere, error) {
 	return getVSphere(c)
 }
+
+// get .vmx file path for a virtual machine
+func getVMXFilePath(vmObject *object.VirtualMachine) (vmxPath string) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var nodeVM mo.VirtualMachine
+	err := vmObject.Properties(ctx, vmObject.Reference(), []string{"config.files"}, &nodeVM)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(nodeVM.Config).NotTo(BeNil())
+
+	vmxPath = nodeVM.Config.Files.VmPathName
+	framework.Logf("vmx file path is %s", vmxPath)
+	return vmxPath
+}
+
+// verify ready node count. Try upto 3 minutes. If count is expected count, return the nodeList
+func verifyReadyNodeCount(client clientset.Interface, expectedNodes int) (nodeList *v1.NodeList) {
+	numNodes := 0
+	for i := 0; i < 31; i++ {
+		nodeList = framework.GetReadySchedulableNodesOrDie(client)
+		Expect(nodeList.Items).NotTo(BeEmpty(), "Unable to find ready and schedulable Node")
+
+		numNodes = len(nodeList.Items)
+		if numNodes == expectedNodes {
+			break
+		}
+		time.Sleep(5 * time.Second)
+	}
+	Expect(numNodes).To(Equal(expectedNodes))
+	return nodeList
+}
+
+// poweroff nodeVM and confirm the poweroff state
+func poweroffNodeVM(nodeName string, vm *object.VirtualMachine) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	framework.Logf("Powering off node VM %s", nodeName)
+
+	_, err := vm.PowerOff(ctx)
+	Expect(err).NotTo(HaveOccurred())
+	err = vm.WaitForPowerState(ctx, vimtypes.VirtualMachinePowerStatePoweredOff)
+	Expect(err).NotTo(HaveOccurred(), "Unable to power off the node")
```

Comment:
Log the error. It's not getting logged anywhere.

Reply:
In case of error, Expect clause will fail and log err as well. Wouldn't that be sufficient?
Eg:
```
    Expected error:
        <*errors.errorString | 0xc421724680>: {
            s: "PersistentVolume vspherepv-m69tb still exists within 30s",
        }
        PersistentVolume vspherepv-m69tb still exists within 30s
    not to have occurred
```

Reply:
Ok, good. This is enough.